apps/prairielearn/src/pages/studentAssessmentInstance/studentAssessmentInstance.ts

```diff
@@ -150,32 +150,29 @@ router.post(
       await processDeleteFile(req, res);
       res.redirect(req.originalUrl);
     } else if (['grade', 'finish', 'timeLimitFinish'].includes(req.body.__action)) {
-      let closeExam: boolean;
       if (req.body.__action === 'grade') {
         if (!res.locals.assessment.allow_real_time_grading) {
           throw new HttpStatusError(403, 'Real-time grading is not allowed for this assessment');
         }
-        closeExam = false;
-      } else if (req.body.__action === 'finish') {
-        closeExam = true;
       } else if (req.body.__action === 'timeLimitFinish') {
         // Only close if the timer expired due to time limit, not for access end
         if (!res.locals.assessment_instance_time_limit_expired) {
           return res.redirect(req.originalUrl);
         }
-        closeExam = true;
-      } else {
-        throw new HttpStatusError(400, `unknown __action: ${req.body.__action}`);
       }
+
+      const isFinishing = ['finish', 'timeLimitFinish'].includes(req.body.__action);
       await assessment.gradeAssessmentInstance({
         assessment_instance_id: res.locals.assessment_instance.id,
         user_id: res.locals.user.user_id,
         authn_user_id: res.locals.authn_user.user_id,
         requireOpen: true,
-        close: closeExam,
-        ignoreGradeRateLimit: false,
+        close: isFinishing,
```

Comment:
Double-check my logic here.

Reply:
This looks good - the outcome of `isFinishing` is the same.